### ISSUE TITLE
S3 Validation Not Accepting References

### ISSUE THREAD

**Issue by jarosser06** [cloudtools/troposphere#846]
The S3 validation for Troposphere now breaks when you try to reference a parameter to specify S3 Bucket ACLs. This is caused by the validation() function on the S3 Bucket Object that checks for a specific string. The validation should allow for Intrinsic Functions as well as just the static strings.

Affected Code:
https://github.com/cloudtools/troposphere/blob/master/troposphere/s3.py#L302-L307

Example Breaking test:
```python
from troposphere import Output, Ref, Template, Parameter
from troposphere.s3 import Bucket, PublicRead


t = Template()
bucket_acl = t.add_parameter(Parameter(
    "BucketACL",
    Type="String",
    Default="Private",
    Description="My Test Bucket ACL",
))

t.add_description(
    "AWS CloudFormation Sample Template S3_Bucket: Sample template showing "
    "how to create a publicly accessible S3 bucket. "
    "**WARNING** This template creates an Amazon S3 Bucket. "
    "You will be billed for the AWS resources used if you create "
    "a stack from this template.")

s3bucket = t.add_resource(Bucket(
    "S3Bucket",
    AccessControl=Ref(bucket_acl),
))

t.add_output(Output(
    "BucketName",
    Value=Ref(s3bucket),
    Description="Name of S3 bucket to hold website content"
))

print(t.to_json())
```

The above code would result in the following error:
```
Traceback (most recent call last):                                                                                                                                                                                                          
  File "foo.py", line 34, in <module>                                                                                                                                                                                                       
    print(t.to_json())                                                                                                                                                                                                                      
  File "/Users/jim/.virtualenvs/troposphere-bug/lib/python2.7/site-packages/troposphere/__init__.py", line 617, in to_json                                                                                                                  
    return json.dumps(self.to_dict(), indent=indent,                                                                                                                                                                                        
  File "/Users/jim/.virtualenvs/troposphere-bug/lib/python2.7/site-packages/troposphere/__init__.py", line 614, in to_dict                                                                                                                  
    return encode_to_dict(t)                                                                                                                                                                                                                
  File "/Users/jim/.virtualenvs/troposphere-bug/lib/python2.7/site-packages/troposphere/__init__.py", line 65, in encode_to_dict                                                                                                            
    props[name] = encode_to_dict(prop)                                                                                                                                                                                                      
  File "/Users/jim/.virtualenvs/troposphere-bug/lib/python2.7/site-packages/troposphere/__init__.py", line 65, in encode_to_dict
    props[name] = encode_to_dict(prop)
  File "/Users/jim/.virtualenvs/troposphere-bug/lib/python2.7/site-packages/troposphere/__init__.py", line 56, in encode_to_dict
    return encode_to_dict(obj.to_dict())
  File "/Users/jim/.virtualenvs/troposphere-bug/lib/python2.7/site-packages/troposphere/__init__.py", line 234, in to_dict
    self.validate()
  File "/Users/jim/.virtualenvs/troposphere-bug/lib/python2.7/site-packages/troposphere/s3.py", line 307, in validate
    ', '.join(self.access_control_types)))
ValueError: AccessControl must be one of "Private, PublicRead, PublicReadWrite, AuthenticatedRead, BucketOwnerRead, BucketOwnerFullControl, LogDeliveryWrite"
```

**Comment by martinb3** [comment]
Previously the validation only considered strings, e.g. `isinstance(kwargs['AccessControl'], basestring)` was in the condition. It was removed in #844 by @markpeek.

**Comment by markpeek** [comment]
Thank you for the issue. From my testing 1.9.6 and 2.0.0 both fail with ValueError when using a Ref(). Your use case is valid and I just pushed a fix onto master (82f0bee) for you to test.

**Comment by jarosser06** [comment]
Thanks @markpeek I verified this is no longer an issue.
